src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -224,14 +224,14 @@ UCS_CLASS_DEFINE_DELETE_FUNC(uct_ud_mlx5_ep_t, uct_ep_t);
  * Generic inline+iov post-send function
  * The caller should check that header size + sg list would not exceed WQE size.
  */
-static UCS_F_ALWAYS_INLINE ucs_status_t
-uct_ud_mlx5_ep_inline_iov_post(uct_ep_h tl_ep, uint8_t am_id,
-                               /* inl. header */ const void *header, size_t header_size,
-                               /* inl. data */   const void *data, size_t data_size,
-                               /* iov data */    const uct_iov_t *iov, size_t iovcnt,
-                               uint32_t packet_flags, uct_completion_t *comp,
-                               unsigned stat_ops_counter, unsigned stat_bytes_counter,
-                               const char *func_name)
+static UCS_F_ALWAYS_INLINE ucs_status_t uct_ud_mlx5_ep_inline_iov_post(
+        uct_ep_h tl_ep, uint8_t am_id,
+        /* inl. header */ const void *header, size_t header_size,
+        /* inl. data */ const void *data, size_t data_size, int is_iov,
```

Comment:
i think it's not so good to pass "iov" as "data" because it may be confusing,
actually, replace "header, header_size" and "data, data_size" by "inl_iov, inl_iovcnt"
for example, uct_ud_mlx5_ep_am_zcopy will pass "inl_iov[0].buffer=header,inl_iov[0].length=header_len,inl_iovcnt=1"
and uct_ud_mlx5_ep_short_common will pass "inl_iov[0]=header. inl_iov[1]=data,inl_iovcnt=2"
just need to check the compiler can optimize and unroll copy loop when iovcnt=1 and 2
WDYT?